### PR TITLE
feat(Other): Add final design changes for MAX32572 related files

### PR DESCRIPTION
**Pull request by sihyung-maxim**
### Description

This PR adds the latest MAX32572 library files. These files match what's in the `dev/me13b_me55` branch.

I've also made changes to the TMR enum values - meaning the `MXC_` prefix was added while keeping the old names. Many names that did not follow the SDK convention but were left in the dust were finally updated for all parts during the development of the ME55 and ME13B SDK files. For example:
image

### Checklist Before Requesting Review

- [ ] PR Title follows correct guidelines.
- [ ] Description of changes and all other relevant information.
- [ ] (Optional) Link any related GitHub issues [using a keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)
- [ ] (Optional) Provide info on any relevant functional testing/validation.  For API changes or significant features, this is not optional.

### CODE REVIEW

File: Libraries/CMSIS/Device/Maxim/MAX32572/Include/max32572.h [hunk 1]
```diff
@@ -643,34 +683,33 @@ typedef enum {
...
-#define MXC_SPI1 ((mxc_spi_regs_t *)MXC_BASE_SPI1)
-#define MXC_BASE_SPI2 ((uint32_t)0x400BE000UL)
-#define MXC_SPI2 ((mxc_spi_regs_t *)MXC_BASE_SPI2)
+#define MXC_BASE_SPI0                   ((uint32_t)0x40046000UL)
+#define MXC_SPI0                        ((mxc_spi_regs_t *)MXC_BASE_SPI0)
+#define MXC_BASE_SPI1                   ((uint32_t)0x40047000UL)
+#define MXC_SPI1                        ((mxc_spi_regs_t *)MXC_BASE_SPI1)
+// SPI2 does not exist in the MAX32572 (to match instance addressing with MAX32570B)
+#define MXC_BASE_SPI3                   ((uint32_t)0x400BE000UL)
+#define MXC_SPI3                        ((mxc_spi_regs_t *)MXC_BASE_SPI3)
```
Comment (sihyung-maxim) [inline Libraries/CMSIS/Device/Maxim/MAX32572/Include/max32572.h:692]: As mentioned earlier, the ME55 is a subset of the ME13B - with a few features removed on the ME55. SPI2 peripheral location is not supported on the ME55, but the numbering is intentional to keep the SPI peripheral addresses and numbering consistent between the ME13B and ME55.

File: Libraries/PeriphDrivers/Source/CTB/aes_reva.svd [hunk 2]
```diff
@@ -1,33 +0,0 @@
-<?xml version="1.0" encoding="utf-8" standalone="no"?>
```
Comment (sihyung-maxim) [inline Libraries/PeriphDrivers/Source/CTB/aes_reva.svd:1]: This file is not used for any of the parts.

**Review by cwarer1 — APPROVED** [review]

File: Libraries/CMSIS/Device/Maxim/MAX32572/Source/GCC/max32572.ld [hunk 3]
```diff
@@ -73,77 +57,40 @@ SECTIONS {
...
-        *(EXCLUDE_FILE(*crtend?.o *crtend.o) .ctors)
-        *(SORT(.ctors.*))
-        *(.ctors)
-
-        /* .dtors */
-        *crtbegin.o(.dtors)
-        *crtbegin?.o(.dtors)
-        *(EXCLUDE_FILE(*crtend?.o *crtend.o) .dtors)
-        *(SORT(.dtors.*))
-        *(.dtors)
```
Comment (Jake-Carter) [inline Libraries/CMSIS/Device/Maxim/MAX32572/Source/GCC/max32572.ld:88]: ctors and dtors are needed for C++ support, any idea why they were removed?

File: Libraries/Boards/MAX32572/BCB/Include/board.h [hunk 4]
```diff
@@ -0,0 +1,85 @@
+/******************************************************************************
+ * Copyright (C) 2022 Maxim Integrated Products, Inc., All rights Reserved.
```
[the remark this answers is not in the document]
Comment (sihyung-maxim) [reply Libraries/Boards/MAX32572/BCB/Include/board.h:2]: Nice catch.

File: Libraries/Boards/MAX32572/BCB/board.mk [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+ ###############################################################################
+
+ifeq "$(BOARD_DIR)" ""
+$(error BOARD_DIR must be set)
+endif
+
+# PROJ_CFLAGS+=-DRO_FREQ=46000000
+# PROJ_CFLAGS+=-DCRYPTO_FREQ=46000000
+
+#PROJ_CFLAGS+=-DPeripheralClock=60000000
```
[the remark this answers is not in the document]
Comment (sihyung-maxim) [reply Libraries/Boards/MAX32572/BCB/board.mk:43]: Removed.

File: Libraries/CMSIS/Device/Maxim/MAX32572/Source/GCC/startup_riscv_max32572.S [hunk 6]
```diff
@@ -1,19 +1,39 @@
-// Copyright 2017 ETH Zurich and University of Bologna.
```
[the remark this answers is not in the document]
Comment (sihyung-maxim) [reply Libraries/CMSIS/Device/Maxim/MAX32572/Source/GCC/startup_riscv_max32572.S:1]: Yep, should be a portions copyright.

File: Libraries/CMSIS/Device/Maxim/MAX32572/Source/GCC/max32572.ld (hunk 3 above)
Comment (sihyung-maxim) [reply Libraries/CMSIS/Device/Maxim/MAX32572/Source/GCC/max32572.ld:88]: This was created roughly 1-2 years ago, and has been kept separate/out of sync with the main as development was on and off.

**Review by lorne-maxim — APPROVED** [review]

**Pull request merged**